core/src/validation/engine.ts

```diff
@@ -103,21 +109,52 @@ export class ValidationEngine extends EventEmitter {
       ? await this.dal.mitigationService.list(model.id)
       : [];
 
+    const threatSuggestions = model?.id
+      ? await this.dal.suggestionService.listThreatSuggestions(model.id)
+      : [];
+    const controlSuggestions = model?.id
+      ? await this.dal.suggestionService.listControlSuggestions(model.id)
+      : [];
+
     // Rules
     const componentRules = this.rules.filter(
       (rule) => rule.type === "component"
     );
     const modelRules = this.rules.filter((rule) => rule.type === "model");
     const results: ValidationResult[] = [];
 
+    const ruleArgs = {
+      model,
+      threats,
+      controls,
+      mitigations,
+      threatSuggestions,
+      controlSuggestions,
+    };
+
     // Validate model
     for (const rule of modelRules) {
       if (!isModelValidation(rule)) {
         continue;
       }
 
+      if (rule.conditionalRules && rule.conditionalRules.length > 0) {
+        const conditions = rule.conditionalRules.map(async (condition) => {
+          return await condition(ruleArgs);
+        });
+        const results = await Promise.all(conditions);
+        console.log(`Conditional results: ${results}`);
```

Comment:
Should not use console.log, use the logger instead (look for `getLogger` in the code)